### PR TITLE
Fix Android sometimes missing callbacks

### PR DESCRIPTION
**Pull request by mattijsf**
I noticed that callbacks on Android don't always work. Here's some example code:

```js
const time = Date.now();
let hasResult = false;
const zoom1 = this._map.getZoom();
const zoom2 = this._map.getZoom();
const zoom3 = this._map.getZoom();
console.log(`Request Zoom: ${time}`);
Promise.all([zoom1, zoom2, zoom3]).then(values => {
  hasResult = true;
  console.log(`Received Zoom: ${time}`);
  this.setState({zoom: values[0]});
});
setTimeout(() => {
  if (!hasResult) {
    console.log(`Request Zoom: ${time} timed-out`);
  }
}, 1000);
```

In which I added the following in `_runNativeCommand` inside `MapView.js` for debugging purposes.
```
console.log(``${methodName}: ${callbackID}`)
```

This is the output:

```js
getZoom: 1561558978423
getZoom: 1561558978424
getZoom: 1561558978424
Request Zoom: 1561558978423
Request Zoom: 1561558978423 timed-out
```
_Note it's timing sensitive so this might only occur in faster environments_

Two issues:

- `callbackID` is not unique enough. It only uses `Date.now()` with the chance the callback method gets overridden in `_callbackMap` within the same millisecond.

- Callbacks are dispatched via [EventDispatcher](https://github.com/react-native-mapbox-gl/maps/blob/master/android/rctmgl/src/main/java/com/mapbox/rctmgl/components/AbstractEventEmitter.java#L46) which by default [merges](https://github.com/facebook/react-native/blob/1151c096dab17e5d9a6ac05b61aacecd4305f3db/ReactAndroid/src/main/java/com/facebook/react/uimanager/events/Event.java#L59-L75) events with the same `eventName (string)`, `viewId (int)`, and `coalescingKey (int)`

In case of callbacks such as `getZoom` / `getCenter` etc these values are:

- `eventName` = `rct.mapbox.map.androidcallback`

- `viewId` = `mapView.getID()`

- `coalescingKey` = 0

Which is fine for most events but for callbacks the same `eventName` `rct.mapbox.map.androidcallback` is re-used for different JS callback methods / callback-references.

Basically means that there is a high likelihood that callbacks are ignored when executed rapidly.

### CODE REVIEW

File: javascript/components/MapView.js [hunk 1]
```diff
@@ -479,7 +481,8 @@ class MapView extends NativeBridgeComponent {
 
     if (isAndroid()) {
       return new Promise(resolve => {
-        const callbackID = `${Date.now()}`;
+        callbackIncrement++;
```
Comment (mfazekas) [inline javascript/components/MapView.js]: FWIW we do have ```callbackID = `${Date.now()}`;```  in NativeBridgeComponent.js

File: javascript/components/MapView.js [hunk 2]
```diff
@@ -28,6 +28,8 @@ export const NATIVE_MODULE_NAME = 'RCTMGLMapView';
 
 export const ANDROID_TEXTURE_NATIVE_MODULE_NAME = 'RCTMGLAndroidTextureMapView';
 
+let callbackIncrement = 0;
```
Comment (mfazekas) [inline javascript/components/MapView.js]: I'd consider adding this as static to NativeBridgeComponent. So `Camera` will be fixed as well.

Comment (mattijsf) [reply javascript/components/MapView.js]: I don't think `<Camera />` is currently using any of the `NativeBridgeComponent` functionality as it doesn't register the `_onAndroidCallback` in `<RCTMGLCamera ... />`

But I did oversee the code inside `NativeBridgeComponent` it's subject to the same problem if used.

If possible I think it is better to either: 
- Remove `NativeBridgeComponent` as it is not used (I think)
- Migrate the `_runNativeCommand` code from `MapView` to `NativeBridgeComponent` as `MapView` also extends it and the code is very similar. I think the `MapView._runNativeCommand` code is better as it also queues methods that are awaiting `nativeRef`.

What do you think?

Comment (mfazekas) [reply javascript/components/MapView.js]: Yes, you're right it seems `NativeBridgeComponent` is not used by `Camera`. 

Please use whichever solution you think is better, i'm not 100% i understand exactly what's going on there.

Comment (mfazekas) [reply javascript/components/MapView.js]: I think what went on is that the expression branch did the refactoring of [extracting NativeBridgeComponent from MapView](https://github.com/nitaliano/react-native-mapbox-gl/commit/83f74581c26115e08ddfa0d09a85dfde214fa5ab#diff-eeef59bd5c74819917627d517c16ae03R40). In the normal code MapView got a fix of waiting for [_nativeRef if needed](https://github.com/nitaliano/react-native-mapbox-gl/commit/ffcbf2d2e4419e7a691d0a4de8ab3fe4d0bfdf56).

And i missmerged the two. I think it's reasonable to move down event handler infra to NativeBridgeComponent from MapView, so i'd vote for the second option, but you have better understanding of the code, so feel free to decide.

Comment (mattijsf) [reply javascript/components/MapView.js]: I agree, it's a good abstraction layer for any future (Android) component that needs method callbacks. I'll work on it in the coming days. I have some other work that requires my attention atm.

**Comment by mattijsf** [comment]
@mfazekas do you want to have a final look? It is ready to be merged.

**Comment by mattijsf** [comment]
Just noticed I still need to remove NativeBridgeComponent from Camera. Should be good to go after that. I'll do that this evening.
